### ISSUE TITLE
Impossible to update composer with the current setup

### ISSUE THREAD

**Issue by francislavoie** [p2made/yii2-sb-admin-theme#8]
I'm not exactly sure what you did, but I can't update this lib right now, but I really need to, to get font-awesome 4.7.0. 

I get stuff like this.

    Problem 1
      - Installation request for p2made/yii2-sb-admin-theme 1.5.9@dev -> satisfiable by p2made/yii2-sb-admin-theme[1.5.9].
      - p2made/yii2-sb-admin-theme 1.5.9 requires p2made/yii2-p2y2-things dev-master -> satisfiable by p2made/yii2-p2y2-things[dev-master].
      - p2made/yii2-p2y2-things dev-master requires rmrevin/yii2-fontawesome >=2.17.1 -> satisfiable by rmrevin/yii2-fontawesome[2.17.1, dev-master].
      - rmrevin/yii2-fontawesome 2.17.1 requires fortawesome/font-awesome ~4.7 -> satisfiable by fortawesome/font-awesome[v4.7.0].
      - rmrevin/yii2-fontawesome dev-master requires fortawesome/font-awesome ~4.7 -> satisfiable by fortawesome/font-awesome[v4.7.0].
      - Conclusion: don't install fortawesome/font-awesome v4.7.0

I tried 1.5.8 but I get this

    Problem 1
      - The requested package p2made/yii2-sb-admin-theme 1.5.8 exists as p2made/yii2-sb-admin-theme[1.5.9, dev-SB-Admin-orig, dev-master, v1.5.7] but these are rejected by your constraint.

:cry:

**Comment by francislavoie** [comment]
Sidenote: do really this many things need to be required as bower-assets? It takes SOOOOO long every time I need to update anything while requiring this lib. It needs to do 6 API calls to github per dependency. Takes forever.

**Comment by p2made** [comment]
I've been struggling with dependency resolution, & have resorted to usign `dev-master` as the version. I suggest you try the same. I'm also beginning to suspect that dependency resolution why many Composer projects specify `dev-master` as the version to use.

Yes, I agree that updates take a while. The bower dependancies are in `p2made/yii2-p2y2-things` which is a large collection of assets (including FontAwesome). I could bundle them as libraries but that defeats the point of dependancy management. You could fork my projects & cut the dependancies down to only those ones you need.

**Comment by francislavoie** [comment]
I've tried `dev-master`, it failed. The problem is that my project uses `minimum-stability` `stable`, and I'm not going to change that. There's too many other dependencies that it would break by pulling dev versions of those.

See https://github.com/composer/composer/issues/6107, it seems that I'm not the only one with these problems.

I guess I might just need to fork it... 😢

**Comment by andrew-svirin** [comment]
Hello,
You can copy 3 Assets Files and 1 JS + 1 CSS file + templates folder in your own theme and be happy do not use this hard uploading library for you project.
```
Loading composer repositories with package information
Updating dependencies (including require-dev)                
  - Removing p2made/yii2-p2y2-things-demo (1.15.0)                  
  - Removing p2made/yii2-p2y2-things (1.15.0)
  - Removing bower-asset/datatables-responsive (2.1.1)
  - Removing bower-asset/jszip (v3.1.3)
  - Removing bower-asset/pdfmake (0.1.24)
  - Removing bower-asset/animate.css (3.5.2)
  - Removing bower-asset/bootstrap-social (5.1.1)
  - Removing bower-asset/bootstrap-sweetalert (1.0.1)
  - Removing bower-asset/bootswatch (v3.3.7)
  - Removing bower-asset/datatables-autofill (2.1.3)
  - Removing bower-asset/datatables-buttons (1.2.4)
  - Removing bower-asset/datatables-colreorder (1.3.2)
  - Removing bower-asset/datatables-fixedcolumns (3.2.2)
  - Removing bower-asset/datatables-fixedheader (3.1.2)
  - Removing bower-asset/datatables-keytable (2.2.0)
  - Removing bower-asset/datatables-rowreorder (1.2.0)
  - Removing bower-asset/datatables-scroller (1.4.2)
  - Removing bower-asset/datatables-select (1.2.1)
  - Removing bower-asset/datatables (1.10.13)
  - Removing bower-asset/ekko-lightbox (v4.0.2)
  - Removing bower-asset/flag-icon-css (2.8.0)
  - Removing bower-asset/flot.tooltip (0.9.0)
  - Removing bower-asset/gmaps (0.4.24)
  - Removing bower-asset/holderjs (v2.9.4)
  - Removing bower-asset/isotope (v3.0.1)
  - Removing bower-asset/masonry (v4.1.1)
  - Removing bower-asset/outlayer (v2.1.0)
  - Removing bower-asset/jquery-migrate (3.0.0)
  - Removing bower-asset/jquery.fitvids (v1.2.0)
  - Removing bower-asset/lightbox2 (v2.9.0)
  - Removing bower-asset/metismenu (v2.6.1)
  - Removing bower-asset/moment-timezone (0.5.11)
  - Removing bower-asset/qunit (2.1.1)
  - Removing bower-asset/raphael (v2.2.7)
  - Removing bower-asset/eve-raphael (v0.5.0)
  - Removing moontoast/math (1.1.1)
  - Removing philippfrenzel/yii2fullcalendar (3.0.3)
  - Removing bower-asset/fullcalendar (v3.1.0)
  - Removing bower-asset/moment (2.17.1)
  - Removing ramsey/uuid (3.5.2)
  - Removing fxp/composer-asset-plugin (v1.2.2)
  - Removing bower-asset/font-awesome (v4.7.0)
  - Removing bower-asset/bootstrap-switch (v3.3.2)
  - Removing bower-asset/jquery-countto (v1.2.0)
  - Removing bower-asset/ev-emitter (v1.0.3)
  - Removing bower-asset/get-size (v2.0.2)
  - Removing bower-asset/fizzy-ui-utils (v2.0.3)
  - Removing bower-asset/desandro-matches-selector (v2.0.1)
  - Removing bower-asset/sweetalert (v1.1.3)
  - Removing bower-asset/wow (1.1.2)
  - Removing paragonie/random_compat (v2.0.4)
  - Removing p2made/yii2-sb-admin-theme (1.5.9)
  - Installing onokumus/metismenu (v2.6.1)
    Downloading: 100%  
```
This is my composer log.

**Comment by francislavoie** [comment]
@p2made Are you sure there's nothing you can do to fix this? No combination of version settings work with composer to update your lib. It was working just fine before, why did it need to change? I'm not sure exactly which change broke it either, it's a bit hard to track down.

**Comment by p2made** [comment]
I am doing what I can, including picking through all the dependancies in `p2made/yii2-p2y2-things`, & some of the bower dependancies have now been replaced with composer ones. At last run `^1.5` works as version constraint for `p2made/yii2-sb-admin-theme`.

**Comment by francislavoie** [comment]
Ok I'll give that a shot and report back. Thanks

**Comment by francislavoie** [comment]
Alright, looks good, at a glance it seems to work for me. I'll have to do some more extensive testing on Monday. I did notice one problem though, see my PR https://github.com/p2made/yii2-p2y2-things/pull/4, easy fix.

**Comment by francislavoie** [comment]
Really though, as frustrated as I might have sounded, I really highly appreciate all the work you've put into this. It's awesome that I have all these assets at my disposal in a pretty easy to use way. Sorry if I wasn't as friendly as I could've been. Thanks a ton for your work!!

😀

**Comment by francislavoie** [comment]
@p2made actually, noticed one thing that seems weird - when I did a composer update it moved a bunch of stuff into `/components` of my project, outside of `/vendor`... any ideas why that'd happen? I don't actually use any of the libs that got thrown in there, but it was weird. Stuff like flag-icon-css, requireJS, animate.css, etc. I didn't note all of them before removing, sorry.

Edit: I had it stashed, [here's the full list](http://i.imgur.com/3mdAXj9.png)

**Comment by p2made** [comment]
I hadn't noticed that at the time (it was past 3 am for me), but same has happened for me. Curiously they all ones where I'd changed from bower dependancies to composer ones, & all those that I've checked so far appear to be copies (or partial copies). I'm wondering whether it might be because I'd added `robloach/component-installer` to the dependancies for `p2made/yii2-p2y2-things` (added as a possible way of further removing dower deps). Right now I'm running `composer update` on a clean Yii2 install to see what happens without component-installer.

**Comment by francislavoie** [comment]
See here https://github.com/RobLoach/component-installer#installation-directory you can configure the components dir, it should probably change to be within `p2made/yii2-p2y2-things` or something.

**Comment by p2made** [comment]
I've now tried a bunch more things, including a brand new clean Yii2 to play in, & emptying the composer cache (`~/.composer/cache`). Still copies those files to `project-dir/components`.

Re Raphael... that was an oddity for a while due to a dependancy issue of its own. Because of that I'd been using a copy of Raphael in `yii2-p2y2-things/assets/lib`. That issue seems to have been resolved, so I've removed that copy & updated the path spec in the asset accordingly. The versions specifies within my assets apply (with few exceptions) only when pulling the CSS &/or JS from CDNs. With raphael, `p2made/yii2-p2y2-things` is installing v2.2.7, which is also on the CDN, but inside the JS file is still says 2.2.0.
